private/pkg/git/repository.go

```diff
@@ -153,41 +165,12 @@ func (r *repository) CurrentBranch() string {
 func (r *repository) ForEachCommit(f func(Commit) error, options ...ForEachCommitOption) error {
 	var config forEachCommitOpts
 	for _, option := range options {
-		if err := option(&config); err != nil {
-			return err
-		}
+		option(&config)
 	}
-	var startCommit Commit
-	if config.start == nil {
-		// if no custom start point is set, use HEAD from the default branch
-		var err error
-		startCommit, err = r.HEADCommit(r.DefaultBranch())
-		if err != nil {
-			return fmt.Errorf("get head commit for default branch %q: %w", r.DefaultBranch(), err)
-		}
-	} else {
-		switch config.start.refType {
-		case refTypeHash:
-			commitID, err := NewHashFromHex(config.start.refName)
-			if err != nil {
-				return fmt.Errorf("new hash from %s: %w", config.start.refName, err)
-			}
-			startCommit, err = r.objectReader.Commit(commitID)
-			if err != nil {
-				return fmt.Errorf("read commit %s: %w", commitID.Hex(), err)
-			}
-		case refTypeBranch:
-			branch := normalpath.Unnormalize(config.start.refName)
```

Comment:
This turned out to fail in Windows tests. I'm a bit confused there, afaik windows creates branches refs `"foo/bar"`  as `foo\bar`, but it failed finding the branch. Did you remember if you see something similar in the first implementations of the git pkg @saquibmian ?

Reply:
Removing the `Unnormalize` line fixes the test, but that's what's confusing to me, isn't the branch supposed to be created normalized with backwards slashes?

Reply:
It's indeed related to normalization. Switched the usages of `path.Join` and other `path.` funcs to `normalpath.` pkg, which internally handles gracefully the OS separators.